scvi/inference/inference.py

```diff
@@ -71,3 +80,221 @@ def train(self, n_path=10, n_epochs=50, **kwargs):
             super().train(n_epochs, params=self.params, **kwargs)
 
         return min(self.history["ll_test_set"])
+
+
+def auto_tuned_scvi_parameters(
+    gene_dataset: GeneExpressionDataset,
+    model_class: VAE = VAE,
+    trainer_class: Trainer = UnsupervisedTrainer,
+    model_specific_kwargs: dict = None,
+    trainer_specific_kwargs: dict = None,
+    train_func_specific_kwargs: dict = None,
+    space: dict = None,
+    use_batches: bool = False,
+    max_evals: int = 10,
+    verbose: bool = True,
+) -> (Type[Trainer], Trials):
+    """Perform automatic hyperparameter optimization of an scVI model
+    and return best model and hyperopt Trials object.
+    Trial object contains hyperparameter space and loss history for each trial.
+    We provide a default hyperparameter search space (see source code),
+    but we recommend the user to build a custom one for each application.
```

Comment:
If at all possible, we should find a single set of bounds for the hyperparameter search space that work for all applications. It's important for three reasons. 1) Most users will have no idea how to customize the search space, 2) It makes it hard to compare results if we are using multiple procedures for selecting hyperparameters, and 3) reviewers for our papers have been uncomfortable with procedures that have free parameters to set.

How about by default we use a search space that is so large that it will work for all applications, but we give the user the option to override those defaults if he/she happens to know tighter bounds?